app/models/article.rb

```diff
@@ -19,16 +19,22 @@ class Article < ApplicationRecord
   delegate :name, to: :user, prefix: true
   delegate :username, to: :user, prefix: true
 
-  belongs_to :user
-  belongs_to :organization, optional: true
   # touch: true was removed because when an article is updated, the associated collection
   # is touched along with all its articles(including this one). This causes eventually a deadlock.
   belongs_to :collection, optional: true
+  belongs_to :organization, optional: true
+  belongs_to :user
 
   counter_culture :user
   counter_culture :organization
 
-  has_many :comments, as: :commentable, inverse_of: :commentable
+  has_many :buffer_updates, dependent: :destroy
+  has_many :comments, as: :commentable, inverse_of: :commentable, dependent: :nullify
```

Comment:
a comment belonging to an article should become an orphan when the article is destroyed

Reply:
Are there places that chain methods unto this (e.g. `comment.commetable.title` or whatever) that may need extra guards?

Reply:
Not really, we already render comments when they don't have a commentable anymore: https://github.com/forem/forem/pull/5199

Reply:
@citizen428 we have already fixed most of those thanks to Honeybadgers and orphaned comments already existing.

Reply:
Thanks @rhymes and @mstruve, it's just one of these cases where I'd rather ask one time too many.